### PR TITLE
MODINREACH-239 - INN-Reach Transaction Detail View Action Menu (Patron hold) - Return Item API

### PR DESCRIPTION
**Pull request by oliinyko**
## Purpose
Action menu items provide both a shortcut to perform some circulation actions while viewing transaction details, but also resolve any state inconsistencies between mod-inn-reach transactions and associated FOLIO circulation records (eg. a loan item has been returned but the IN_TRANSIT message has not been sent to the central server, or an item has been cleared from the hold shelf without circulating and not RETURN_UNCIRCULATED message has been sent).

JIRA: [MODINREACH-239](https://issues.folio.org/browse/MODINREACH-239)

## Approach
- Add new endpoint to OpenApi schema and module descriptor
- Add logic to return patron hold item
- Add tests

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by dmtkachenko — APPROVED** [review]

File: src/main/java/org/folio/innreach/controller/InnReachTransactionController.java [hunk 1]
```diff
@@ -41,6 +41,13 @@
     return ResponseEntity.ok(innReachTransaction);
   }
 
+  @Override
+  @PostMapping("/{id}/patronhold/return-item/{servicePointId}")
```
Comment (maryna-steshenko) [inline src/main/java/org/folio/innreach/controller/InnReachTransactionController.java:45]: Should this endpoint also contain servicePointId?

File: src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java [hunk 2]
```diff
@@ -262,6 +261,31 @@ public InnReachTransactionDTO cancelPatronHold(UUID transactionId, CancelPatronH
...
+
+    log.info("Attempting to return item of Patron Hold transaction {}", transactionId);
+
+    var loanId = transaction.getHold().getFolioLoanId();
+    var requestId = transaction.getHold().getFolioRequestId();
+    if (loanId != null) {
+      var loan = loanService.getById(loanId);
+      if (loanService.isOpen(loan)) {
+        log.info("Checking-in item for transaction associated with an open loan");
+        loanService.checkInItem(transaction, servicePointId);
```
Comment (maryna-steshenko) [inline src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java:277]: Shouldn't IN_TRANSIT message be sent as well?

File: src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java [hunk 3]
```diff
@@ -262,6 +261,31 @@ public InnReachTransactionDTO cancelPatronHold(UUID transactionId, CancelPatronH
...
+      }
+    } else {
+      var request = requestService.findRequest(requestId);
+      if (requestService.isCanceledOrExpired(request)) {
+        log.info("Checking-in item for transaction associated with a closed loan and canceled request");
+        loanService.checkInItem(transaction, servicePointId);
```
Comment (maryna-steshenko) [inline src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java:285]: Shouldn't the RETURN_UNCIRCULATED message be sent?

Comment (oliinyko) [reply src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java:285]: This logic will trigger Check-In kafka event and the message will be sent when we receive it

File: src/main/java/org/folio/innreach/controller/InnReachTransactionController.java (hunk 1 above)
Comment (oliinyko) [reply src/main/java/org/folio/innreach/controller/InnReachTransactionController.java:45]: I assume yes, all action menu endpoints for patron hold where check-in logic is applied contain a servicePointId as well

**Review by maryna-steshenko — APPROVED** [review]

File: src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java (hunk 2 above)
Comment (oliinyko) [reply src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java:277]: Check-in leads to loan closure. So we will receive a kafka event for the loan where logic for IN_TRANSIT message will be applied

**Pull request merged**